src/core/backbone_router/bbr_manager.cpp

```diff
@@ -90,18 +111,63 @@ void Manager::HandleMulticastListenerRegistration(const Coap::Message &aMessage,
     // TODO: (MLR) send configured MLR response for Reference Device
     // TODO: (MLR) handle Commissioner Session TLV
     // TODO: (MLR) handle Timeout TLV
-    // TODO: (MLR) handle addresses, updated listener groups, send BMLR.req
+
+    VerifyOrExit(ThreadTlv::FindTlvValueOffset(aMessage, IPv6AddressesTlv::kIPv6Addresses, addressesOffset,
+                                               addressesLength) == OT_ERROR_NONE,
+                 error = OT_ERROR_PARSE);
+    VerifyOrExit(addressesLength % sizeof(Ip6::Address) == 0, error = OT_ERROR_PARSE);
+    VerifyOrExit(addressesLength / sizeof(Ip6::Address) <= kIPv6AddressesNumMax, error = OT_ERROR_PARSE);
```

Comment:
Just for discussion here - how about responding with `kMlrGeneralFailure` for these scenarios to suppress CoAP retransmission from the requester?

Reply:
Checked MATN-TC-21, Step 20, `kMlrGeneralFailure` is one of the three flexible responses.

Reply:
`ST_MLR_GENERAL_FAILURE` is a good idea.

Reply:
changed to return `ST_MLR_GENERAL_FAILURE`.